scripts/library/gemini_client.py

```
@@ -31,29 +31,30 @@ def __init__(self, api_key: Optional[str] = None, cache_path: str = ".gemini_cac
         """Saves the cache to the file."""
-        with self.cache_dictionary_lock:
+        async with self.cache_dictionary_lock:
             cache_to_save = self.cache.copy()
-        with self.cache_file_lock, open(self.cache_path, "w", encoding="utf-8") as f:
-            json.dump(cache_to_save, f, ensure_ascii=False, indent=2)
+        async with self.cache_file_lock:
+            with open(self.cache_path, "w", encoding="utf-8") as f:
+                json.dump(cache_to_save, f, ensure_ascii=False, indent=2)
 
-    def generate_content(self, prompt: str, response_schema, request_id: UUID, **kwargs) -> dict:
+    async def generate_content(self, prompt: str, response_schema, request_id: UUID, **kwargs) -> dict:
```

Pylint (3.3.7)

[convention] Line too long (103/100)

(C0301)